src/components/Layout/Layout.tsx

```diff
@@ -1,14 +1,43 @@
-import React, { ReactNode } from "react";
+import React, { ReactNode, useEffect } from "react";
 import { MainHeader } from "./MainHeader/MainHeader";
 import { Categories } from "./Categories/Categories";
 import { Breadcrumbs } from "./Breadcrumbs/Breadcrumbs";
 import { Footer } from "./Footer/Footer";
+import { AppDispatch } from "../../redux/app/store";
+import { useDispatch } from "react-redux";
+import { useAppSelector } from "../../main/hooks";
+import {
+  getUserProfile,
+  refreshToken,
+} from "../../redux/features/auth/authSlise";
 
 interface IProps {
   children: ReactNode;
 }
 
 export const Layout: React.FC<IProps> = ({ children }) => {
+  const dispatch: AppDispatch = useDispatch();
+  const token = useAppSelector((state) => state.auth.accessToken);
+
+  const isLogin = !!token;
+
+  useEffect(() => {
+    if (isLogin) {
+      dispatch(getUserProfile());
+    }
+  }, [isLogin]);
+
+  useEffect(() => {
+    let interval = 0;
+    if (token) {
+      interval = setTimeout(() => {
+        dispatch(refreshToken());
+      }, 1.9 * 60 * 1000);
+    }
+
+    return () => clearTimeout(interval);
+  }, [token]);
```

Comment:
Let`s discuss it as well

Reply:
@VYaniv , with this useEffect, we trigger the token refresh every 1.9 minutes.